packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -855,10 +857,33 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 this.emit("codeDetailsProposed", proposal.value, proposal);
             }
         });
+        const defaultAction = (op: ISequencedDocumentMessage,sc: SummaryCollection)=> {
+            if(sc.opsSinceLastAck > (this.runtimeOptions.summaryOptions.maxOpsSinceLastSummary ?? 3000)) {
+                this.logger.sendErrorEvent({eventName: "SummaryStatus:Behind"});
+                // unregister default to no log on every op after falling behind
+                // and register summary ack handler to re-register this handler
+                // after successful summary
+                opActions.default = undefined;
+                opActions.summaryAck = summaryAck;
+            }
+        };
+        const summaryAck = (op: ISequencedDocumentMessage,sc: SummaryCollection)=> {
```

Comment:
```suggestion
        const summaryAckAction = (op: ISequencedDocumentMessage,sc: SummaryCollection)=> {
```
More descriptive name like summaryAcktion since in this context it's slightly confusing that this is an action/instruction.